test/integration/common.nix

```diff
@@ -9,16 +16,72 @@
     };
   };
 
-  # Override paretosecurity to use the local codebase
+  # Paretosecurity that uses local codebase
   pareto = {
     pkgs,
     lib,
     ...
   }: {
     services.paretosecurity = {
       enable = true;
-      package = pkgs.callPackage ../../package.nix {inherit lib;};
+      package = paretoLocalPkg {inherit pkgs lib;};
+    };
+  };
+
+  # Paretosecurity with local codebase and patched dashboard URL
+  paretoPatchedDash = {
+    pkgs,
+    lib,
+    ...
+  }: let
+    paretoPkg = paretoLocalPkg {inherit pkgs lib;};
+  in {
+    services.paretosecurity = {
+      enable = true;
+      package = paretoPkg.overrideAttrs (oldAttrs: {
+        postPatch =
+          oldAttrs.postPatch or ""
+          + ''
+            substituteInPlace team/report.go \
+              --replace-warn 'const reportURL = "https://cloud.paretosecurity.com"' \
+                             'const reportURL = "http://dashboard"'
+          '';
+      });
+    };
+  };
+
+  # Dashboard mockup server
+  dashboard = {}: {
+    networking.firewall.allowedTCPPorts = [80];
+
+    services.nginx = {
+      enable = true;
+      virtualHosts."dashboard" = {
+        locations."/api/v1/team/".extraConfig = ''
+          add_header Content-Type application/json;
+          return 200 '{"message": "Linked device."}';
+        '';
+      };
+    };
+  };
+
+  # Common configuration for Display Manager
+  displayManager = {pkgs}: {
+    services.xserver.enable = true;
+    # services.xserver.desktopManager.xfce.enable = true;
+
+    services.displayManager.autoLogin = {
+      enable = true;
+      user = "alice";
     };
+
+    virtualisation.resolution = {
```

Comment:
`virtualisation.resolution` is defined here and also in gnome.nix, is it really required twice?